api/src/controllers/peers.js

```diff
@@ -71,13 +71,14 @@ function PeersControllerFactory (auth, config, log, Peer, connector) {
     static * postResource () {
       const peer = new Peer()
 
-      if (!this.body.hostname || !this.body.limit || !this.body.currency) {
+      if (!this.body.hostname || !this.body.limit || !this.body.currencyCode) {
         throw new InvalidBodyError('At least one of the required fields is missing')
       }
 
       peer.hostname = this.body.hostname.replace(/.*?:\/\//g, '')
-      peer.limit = this.body.limit
-      peer.currency = this.body.currency.toUpperCase()
+      peer.currencyScale = parseInt(this.body.currencyScale) || 9
```

Comment:
I assume 9 is the default currency scale? I think this should be a constant and moved to the top of the file.